hm_9/hm_09.py

```diff
@@ -0,0 +1,64 @@
+term = int(input("Enter a term here: "))
+
+# task_number_one
+
+
+def fib(n):
+    a, b = 0, 1
+    for item in range(n + 1):
+        yield a
+        a, b = b, a + b
+
+
+fib_generator = fib(term)
+for item in fib_generator:
+    pass
+print(item)
+
+# task_number_two
+
+
+class FibIterator:
+    def __init__(self, term_index):
+        self.term_index = term_index
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        a, b = 0, 1
+        for item in range(self.term_index):
```

Comment:
This is a wrong implementation of such an iterator. Your iterator should be able to give you all fibonacci sequence one by one. To test it you can pass your iterator to `next()` function and if you print it one by one you should see all sequence one by one. In your case `FibIterator` cannot be used as iterator but as a method of finding `n` number in the sequence